packages/medusa/src/services/discount.ts

```diff
@@ -688,6 +694,47 @@ class DiscountService extends BaseService {
     })
   }
 
+  async calculateDiscountForLineItem(
+    discountId: string,
+    lineItem: LineItem,
+    cart: Cart
+  ): Promise<number> {
+    let adjustment = 0
+
+    if (!lineItem.allow_discounts) {
+      return adjustment
+    }
+
+    const discount = await this.retrieve(discountId, { relations: ["rule"] })
+
+    const { type, value, allocation } = discount.rule
+
+    const fullItemPrice = lineItem.unit_price * lineItem.quantity
+
+    if (type === DiscountRuleType.PERCENTAGE) {
+      adjustment = (fullItemPrice / 100) * value
```

Comment:
I think we'll also need some rounding here

Reply:
Well spotted 👍